sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/EquivalentExpressions.scala

```diff
@@ -128,10 +131,23 @@ class EquivalentExpressions {
   // There are some special expressions that we should not recurse into all of its children.
   //   1. CodegenFallback: it's children will not be used to generate code (call eval() instead)
   //   2. ConditionalExpression: use its children that will always be evaluated.
-  private def childrenToRecurse(expr: Expression): Seq[Expression] = expr match {
-    case _: CodegenFallback => Nil
-    case c: ConditionalExpression => c.alwaysEvaluatedInputs
-    case other => other.children
+  private def childrenToRecurse(expr: Expression): Seq[Expression] = {
+    val alwaysEvaluated = expr match {
+      case _: CodegenFallback => Nil
+      case c: ConditionalExpression => c.alwaysEvaluatedInputs
+      case other => other.children
+    }
+    if (skipShortcut) {
```

Comment:
`skipShortCut` means we need to handle the shortcut expressions to skip CSE.

Reply:
maybe `skipInShortcut` is a clearer name

Reply:
changed to `skipForShortcut` which aligns with config name